### PR TITLE
Removing LinkCommons type in @fluentui/react-link

### PR DESCRIPTION
**Pull request by khmakoto**
## PR Description

This PR removes the `Commons` type pattern in the types for our `Link` component.

## Related Issue(s)

Fixes part of #22862

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                     |    Baseline (minified/GZIP) |                          PR |                                                                                                                                                                                                Change |
| :------------------------------------------------------------------------------------ | --------------------------: | --------------------------: | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-link</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Link</abbr> | `11.393 kB`<br />`4.609 kB` | `11.394 kB`<br />`4.635 kB` | `1 B` <br />`26 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `179.701 kB`<br />`50.147 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `34.066 kB`<br />`11.089 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/083e54e4374c25be4200c27922c16621431bb2bc'>083e54e4374c25be4200c27922c16621431bb2bc</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22942/perf-test-react-components/Avatar-mount_base.html">1086</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22942/perf-test-react-components/Avatar-mount.html">1131</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22942/perf-test-react-components/Button-mount_base.html">669</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22942/perf-test-react-components/Button-mount.html">676</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22942/perf-test-react-components/FluentProvider-mount_base.html">2225</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22942/perf-test-react-components/FluentProvider-mount.html">2218</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22942/perf-test-react-components/FluentProviderWithTheme-mount_base.html">368</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22942/perf-test-react-components/FluentProviderWithTheme-mount.html">346</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22942/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">276</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22942/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">259</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22942/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">372</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22942/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">350</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22942/perf-test-react-components/MakeStyles-mount_base.html">1868</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22942/perf-test-react-components/MakeStyles-mount.html">1877</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-link/src/components/Link/useLinkStyles.ts [hunk 1]
```diff
@@ -99,15 +99,17 @@ const useStyles = makeStyles({
...
-    state.root.as === 'a' && state.root.href && styles.href,
-    state.appearance === 'subtle' && styles.subtle,
-    state.inline && styles.inline,
-    state.appearance === 'subtle' && state.inline && styles.inlineSubtle,
-    state.root['aria-disabled'] && styles.disabled,
+    root.as === 'a' && root.href && styles.href,
+    appearance === 'subtle' && styles.subtle,
+    inline && styles.inline,
+    appearance === 'subtle' && inline && styles.inlineSubtle,
+    root['aria-disabled'] && styles.disabled,
```
Comment (smhigley) [inline packages/react-components/react-link/src/components/Link/useLinkStyles.ts:112]: Would it make sense to have a single `state.disabled = props.disabled || props.disabledFocusable`? It seems like we don't need to differentiate between disabled and disabledFocusable on `state`.

**Review by smhigley — COMMENTED** [review]
One comment about whether we should have `disabledFocusable` on `LinkState`, otherwise this looks good!

File: packages/react-components/react-link/src/components/Link/useLinkStyles.ts (hunk 1 above)
Comment (khmakoto) [reply packages/react-components/react-link/src/components/Link/useLinkStyles.ts:112]: We cannot, we're setting `state.disabled` as `disabled && !disabledFocusable` when the link renders as a button so the native `disabled` attribute in the button tag is set correctly.

Comment (khmakoto) [reply packages/react-components/react-link/src/components/Link/useLinkStyles.ts]: This is updated now.

**Review by spmonahan — APPROVED** [review]

**Pull request merged**